File: vite-vue-ts-frontend/src/modules/history-operations/types/dto.ts
import type { UserBaseResponse } from "../../users/types/dto";

export type HistoryOperationResponse = {
  id: string;
  createdBy: UserBaseResponse;
  createdAt: number;
  operationType: number;
};

export type SearchResponse = {
  historyOperations: HistoryOperationResponse[];
};

File: golang-backend/internal/handlers/historyoperationhandler/dto.go
package historyoperationhandler

import "github.com/aportela/doneo/internal/handlers/userhandler"

type historyOperationResponse struct {
	ID            string                       `json:"id"`
	CreatedBy     userhandler.UserBaseResponse `json:"createdBy"`
	CreatedAt     int64                        `json:"createdAt"`
	OperationType uint16                       `json:"operationType"`
}

type searchResponse struct {
	HistoryOperations []historyOperationResponse `json:"historyOperations"`
}
